test/Test/Foreign/Object.purs

```diff
@@ -150,6 +153,14 @@ objectTests = do
     quickCheck (O.lookup "1" nums == Just "one"  <?> "invalid lookup - 1")
     quickCheck (O.lookup "2" nums == Nothing     <?> "invalid lookup - 2")
 
+  log "fromFoldableWithIndex & key collision"
+  do
+    let numsMap = M.fromFoldable [Tuple 0 "zero", Tuple 1 "what", Tuple 1 "one"]
+        nums = O.fromFoldableWithIndex show numsMap
+    quickCheck (O.lookup "0" nums == Just "zero" <?> "invalid lookup - 0")
+    quickCheck (O.lookup "1" nums == Just "one"  <?> "invalid lookup - 1")
+    quickCheck (O.lookup "2" nums == Nothing     <?> "invalid lookup - 2")
```

Comment:
Oops, this isn't actually testing key collision on the object, since the `M.fromFoldable` handles that. I'll update.